### PR TITLE
Force loading of packages during merge warnings

### PR DESCRIPTION
**Pull request by theseion**
Fuel is already loaded in Pharo. During load of a newer version Monticello issues an `MCMergeOrLoadWarning` (only exists in Pharo). SmalltalkCI doesn't correctly handle the warning, which results in a mixture of old and new code being loaded.

This fix forces a clean load of the new version, discarding changes in the loaded package.

### CODE REVIEW

**Comment by fniephaus** [comment]
it seems that you are only changing meta information? Or am I missing something?

**Comment by theseion** [comment]
Yeah... Forgot to add the change :)

File: repository/SmalltalkCI-Pharo.package/SmalltalkCIPharo.class/instance/loadProjects.st [hunk 1]
```diff
@@ -0,0 +1,6 @@
+as yet unclassified
```
Comment (fniephaus) [inline repository/SmalltalkCI-Pharo.package/SmalltalkCIPharo.class/instance/loadProjects.st]: Would you be so kind and add this to a category?

**Comment by fniephaus** [comment]
Are you sure that this only needs to be done in Pharo?
@dalehenrich for a second opinion?

**Comment by dalehenrich** [comment]
The "merge or load warning" is Pharo only ... 

The #onWarningLog attribute could be used to allow the load to proceed instead of hard-wiring an exception handler (this is no longer GemStone-specific after my  recent merge --- ought to update README). 

A handler for just the "load or merge Warning" is pretty  specific and won't mask other Warnings that might be more interesting ... Perhaps it would be useful to log the fact that the exception was handled before resuming?

There's an outside chance that it might be desirable to break the build as discussed in Issue #64.

For the long term perhaps we could add a list of Warnings to be handled and resumed in addition to the [fail load on list](https://github.com/hpi-swa/smalltalkCI/issues/64#issuecomment-188481355) and deprecate the #onWarningLog attribute.

File: repository/SmalltalkCI-Pharo.package/SmalltalkCIPharo.class/instance/loadProjects.st (hunk 1 above)
Comment (theseion) [inline repository/SmalltalkCI-Pharo.package/SmalltalkCIPharo.class/instance/loadProjects.st]: Will do.

**Comment by theseion** [comment]
> Are you sure that this only needs to be done in Pharo?

For Fuel yes. In general I expect that MC will issue a `Warning` instead of the more specific `MCMergeOrLoadWarning`.

> A handler for just the "load or merge Warning" is pretty specific and won't mask other Warnings that might be more interesting

True. However, `MCMergeOrLoadWarning` expects `true` when resuming, which is not the default resumption value. So this warning needs special handling.

> Perhaps it would be useful to log the fact that the exception was handled before resuming?

Yes. I had to guess my way to fixing the problem because nothing was visible in the log. It would be nice to see warnings and exceptions, even if they are then being handled.

> There's an outside chance that it might be desirable to break the build as discussed in Issue #64.

I think such a setting is be a good idea (and maybe should be the default). However, for Fuel specifically I must be able to force the load, and there's no way to disable the warnings (unless I patch MC in a `#preLoadDoIt`...)

> For the long term perhaps we could add a list of Warnings to be handled and resumed in addition to the fail load on list and deprecate the #onWarningLog attribute.

While that sounds like a good idea, as I state above I fear that MC in Squeak will issue a `Warning`. So handling of that specific case would entail the same handling for all warnings.
In addition, we would need an option to specify the handling behaviour, as `MCMergeOrLoadWarning` expects a specific value when resumed.

**Comment by theseion** [comment]
I can confirm that Squeak uses `Warning` in `MCVersionLoader>>warnAboutLosingChangesTo:` via `Object>>notify:`

**Comment by fniephaus** [comment]
I've merged this now, so that @theseion can test Fuel.
Could one of you please update #64 or open a new issue to follow up on the MC warning discussion?

**Comment by theseion** [comment]
Thanks. I've opened issue #80.

**Pull request merged**